### DISCUSSION TITLE
Penetration Test

### DISCUSSION THREAD

**Discussion by hayato94087** [vercel/community#1285] [Help] [answered]
### Summary

Is it possible to conduct penetration testing on the environment such as website that was launched with Vercel? I attempted to search for information using various keywords, however, I was unable to find any relevant information. Therefore, I am kindly asking here.　

I provide the URL for the penetration testing offered by AWS below as a reference.
https://aws.amazon.com/security/penetration-testing/?nc1=h_ls

Thank you :)

**Comment by swarnava** [comment, marked as answer]
Hi, Penetration test is only available for Enterprise plan users: https://vercel.com/support/articles/what-s-vercel-s-policy-regarding-load-testing-deployments

**Reply by hayato94087** [reply]
Thank you for your response!
I will consider purchasing an enterprise plan.

**Reply by wereHamster** [reply]
That support article only mentions load testing, not penetration testing. Does the same policy apply for both?

**Reply by swarnava** [reply]
Yes

**Reply by wereHamster** [reply]
@swarnava so for example if a the hired security company loads a website that is hosted on Vercel in a browser, opens devtools and starts inspecting the HTML/JS code and network request (which definitely is a part of any basic penetration testing of a website), without causing any increased or abnormal load on the website, we would be in breach of the fair usage policy?

**Reply by andrewnick** [reply]
We asked this same question through Vercel Support and got the same answer as above. We tried to get further information to why this is the case because the pen testing was going to be similar to what you described. We weren't going to pen test the Vercel infrastructure and weren't going to put the system under any unusual load. 

We kept being referred back to this article and told to get an Enterprise plan setup which would have meant us missing out pen testing window which was very frustrating.

Hopefully we can get more answers here 🤞 or Vercel update their documentation.

**Reply by andrewnick** [reply]
@swarnava are you able to provide further details on our questions above?

**Reply by swarnava** [reply]
It's really hard to tell without knowing the specific cirumcstances. As long as the traffic isn't volumetric in nature, your website will likely perform as expected.

However, our policy is still same. we do not allow penetration testing for Pro, this is limited for Enterprise only customers.  Which we have a standard support workflow to make it smooth.
